pylint/pyreverse/diadefslib.py

```diff
@@ -67,10 +73,14 @@ def _get_levels(self) -> tuple[int, int]:
         return self.anc_level, self.association_level
 
     def show_node(self, node: nodes.ClassDef) -> bool:
-        """True if builtins and not show_builtins."""
-        if self.config.show_builtin:
-            return True
-        return node.root().name != "builtins"  # type: ignore[no-any-return]
+        """Determine if node should be shown based on config."""
+        if node.root().name == "builtins":
+            return self.config.show_builtin  # type: ignore[no-any-return]
+
+        if node.root().name.split(".")[0] in stdlib_module_names:
+            return self.config.show_stdlib  # type: ignore[no-any-return]
```

Comment:
Seems like it's not covered by tests, maybe a small example with a stdlib would be enough ?

Reply:
Tests added